plugins/gateway/kong/plugin.go

```diff
@@ -17,48 +17,66 @@ package main
 
 import (
 	"fmt"
+	"os"
 	"strconv"
 	"strings"
+	"sync"
 	"time"
 
 	"github.com/Kong/go-pdk"
 	"github.com/Kong/go-pdk/server"
 	"github.com/go-openapi/strfmt"
 
-	"github.com/openclarity/apiclarity/plugins/api/client/client"
-	"github.com/openclarity/apiclarity/plugins/api/client/client/operations"
 	"github.com/openclarity/apiclarity/plugins/api/client/models"
 	"github.com/openclarity/apiclarity/plugins/common"
-	"github.com/openclarity/apiclarity/plugins/common/trace_sampling_client"
+	"github.com/openclarity/apiclarity/plugins/common/apiclarity_client"
+)
+
+const (
+	MaxBodySize = 1000 * 1000
 )
 
 var (
-	traceSamplingClient *trace_sampling_client.Client
-	telemetriesAPI      *client.APIClarityPluginsTelemetriesAPI
+	apiclarityClient *apiclarity_client.Client
+	discoveredApis   []string
+	lock             sync.RWMutex
 )
 
 type Config struct {
 	EnableTLS            bool   `json:"enable_tls"`
 	Host                 string `json:"host"`
-	TraceSamplingHost    string `json:"trace_sampling_host"`
 	TraceSamplingEnabled bool   `json:"trace_sampling_enabled"`
+	Token                string `json:"trace_source_token"`
 }
 
 func New() interface{} {
 	return &Config{}
 }
 
 func (conf Config) Access(kong *pdk.PDK) {
-	if conf.TraceSamplingEnabled && traceSamplingClient == nil {
-		_ = kong.Log.Info("Creating trace sampling client")
-		// TODO tls will not work since trace sampling manager is not supporting it currently
-		traceSampling, err := trace_sampling_client.Create(false, conf.TraceSamplingHost, common.SamplingInterval)
+	if conf.TraceSamplingEnabled && apiclarityClient == nil {
+		_ = kong.Log.Info(fmt.Sprintf("Host: '%v'", conf.Host))
+		_ = kong.Log.Info(fmt.Sprintf("TraceSamplingEnabled: '%v'", conf.TraceSamplingEnabled))
+		_ = kong.Log.Info(fmt.Sprintf("Token: '%v'", conf.Token))
+		_ = kong.Log.Info(fmt.Sprintf("SamplingInterval: '%v'", common.SamplingInterval))
+		discoveredApis = []string{}
+		if conf.EnableTLS {
+			if _, err := os.Stat("/etc/traces/certs"); os.IsNotExist(err) {
```

Comment:
"/etc/traces/certs" should go in the common utils.

Reply:
hmmm... why? it's part of the template to patch kong plugin and we can't assume that all plugin will use the same mount path...
But for sure I can put this in a constant or in configuration with default value. 
The goal here is simply to verify that the certificate mount path exists.

Reply:
ok, I see what you mean.  constant CACertFile = "/etc/traces/certs/root-ca.crt" exists in common